cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -230,10 +241,23 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
       for (LocalFile fileToUpload : localFiles) {
         boolean isCacheable = !disableGCSCaching && fileToUpload instanceof CacheableLocalFile;
         String targetFilePath = getPath(isCacheable ? cacheRootPath : runRootPath, fileToUpload.getName());
-        uploadFutures.add(
-          provisionerContext.execute(() -> isCacheable ? uploadCacheableFile(bucket, targetFilePath, fileToUpload) :
-              uploadFile(bucket, targetFilePath, fileToUpload, false))
-            .toCompletableFuture());
+        String targetFilePathWithVersion = getPath(cacheRootPath, cdapVersion[0], fileToUpload.getName());
+
+        if (artifactsCacheablePerCDAPVersion.contains(fileToUpload.getName()) && !disableGCSCaching) {
```

Comment:
Better short circuit the logic to make the intention more clear.
```
if (enableCaching && /* other conditions */) {
}
```
This also explains why I favor positive config. Having:

```
if (enableCaching) {
  // do something
}
```
is better than
```
if (!disableCaching) {
   // do something
}
```

Reply:
ack